apps/chat/src/pages/api/[entitytype]/[...slug].ts

```diff
@@ -19,6 +20,22 @@ import { authOptions } from '@/src/pages/api/auth/[...nextauth]';
 import fetch from 'node-fetch';
 import { Readable } from 'stream';
 
+const getEntityUrlFromSlugs = (
+  dialApiHost: string,
+  req: NextApiRequest,
+): string => {
+  const entityType = getEntityTypeFromPath(req);
+  const slugs = Array.isArray(req.query.slug)
+    ? req.query.slug
+    : [req.query.slug];
+
+  if (!slugs || slugs.length === 0) {
+    throw new OpenAIError(`No ${entityType} path provided`, '', '', '404');
```

Comment:
Maybe 400 error status will be better here?

```suggestion
  if (!slugs || slugs.length === 0) {
    throw new OpenAIError(`No ${entityType} path provided`, '', '', '400');
```